spyder/plugins/completion/providers/languageserver/provider.py

```diff
@@ -772,8 +777,12 @@ def generate_python_config(self):
             for fmt in formatters
         }
 
-        if formatter == 'pylsp_black':
-            formatter_options['pylsp_black']['line_length'] = cs_max_line_length
+        # Setting max line length for formatters
+        # The autopep8 plugin shares the same maxLineLength value with the
+        # pycodestyle one. That's why it's not necessary to set it here.
+        # NOTE: We need to use `black` and not `pylsp_black` because that's
+        # the options' namespace of that plugin.
+        formatter_options['black'] = {'line_length': cs_max_line_length}
```

Comment:
Shouldn't this be something like:
```suggestion
        formatter_options['black']['line_length'] = cs_max_line_length
```
?

Reply:
This doesn't work because the `black` key is not present in the `formatter_options` dict (only `pylsp_black` is). So there are two options about it: I can add that key in the dict comprehension above or we can leave things as they are.

What do you think?

Reply:
Well thought, but those config options end being part of what is sent to the PyLSP because I update the `plugins` dict (which already contains them) with the `formatter_options` one:

https://github.com/spyder-ide/spyder/blob/0c9e015973f2e82d636504972bf772902b4c7e0f/spyder/plugins/completion/providers/languageserver/provider.py#L860